src/components/TradeWidget/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import Widget from 'components/layout/Widget'
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import { Network, TokenDetails } from 'types'
+import { tokenListApi } from 'api'
+import TokenRow from './TokenRow'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faExchangeAlt, faPaperPlane } from '@fortawesome/free-solid-svg-icons'
+
+function _getTokenDetails(symbol: string, networkId: number): TokenDetails {
+  const _networkId = networkId ? networkId : Network.Mainnet // fallback to mainnet
+  return tokenListApi
+    .getTokens(_networkId)
+    .filter(tokenDetails => tokenDetails.symbol == symbol)
+    .reduce((_, tokenDetails) => tokenDetails)
```

Comment:
I don't quite get the intention of all the chaining. This returns the last TokenDetails with a matching symbol.
Just do `tokenListApi.getTokens(_networkId).find(tokenDetails => tokenDetails.symbol == symbol)` or if you really need the last one, `slice().reverse()` first

Reply:
I want to have a single `TokenDetails` instance. Filter returns a new array. I'm using `reduce` to extract the single filtered match, if any.
I opted to not use `.filter(tokenDetails => tokenDetails.symbol == symbol).get(0)` for the case where there's no match and that causes an exception. Although I'm not handling when there's no match outside of the function...

Anyway, do you get the point? 
With this in mind, what would you suggest?

Reply:
```js
tokenListApi.getTokens(_networkId).find(tokenDetails => tokenDetails.symbol == symbol)
```
does exactly what you described, without creating temp arrays